src/evict/evict_page.c

```diff
@@ -353,19 +352,32 @@ __evict_page_dirty_update(WT_SESSION_IMPL *session, WT_REF *ref, bool closing)
 		 * Publish: a barrier to ensure the structure fields are set
 		 * before the state change makes the page available to readers.
 		 */
-		WT_RET(__wt_calloc_one(session, &addr));
-		*addr = mod->mod_replace;
-		mod->mod_replace.addr = NULL;
-		mod->mod_replace.size = 0;
-		ref->addr = addr;
+		if (mod->mod_replace.addr == NULL)
+			ref->addr = NULL;
+		else {
+			WT_RET(__wt_calloc_one(session, &addr));
+			*addr = mod->mod_replace;
+			mod->mod_replace.addr = NULL;
+			mod->mod_replace.size = 0;
+			ref->addr = addr;
+		}
```

Comment:
I don't understand this change. I wouldn't expect to see a `WT_PM_REC_REPLACE` update with `mod->mod_replace.addr == NULL` unless it was a root page (which wouldn't get here).

Reply:
This is the case where no updates were visible (so there is no page image), but updates were saved to the lookaside table.